src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -251,22 +230,18 @@ private long GetLength(in SequencePosition start, in SequencePosition end)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void BoundsCheck(in SequencePosition start, in SequencePosition end)
         {
-            GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out SequenceType type, out int startIndex, out int endIndex);
-
+            int startIndex = GetIndex(start);
+            int endIndex = GetIndex(end);
             object startObject = start.GetObject();
             object endObject = end.GetObject();
+            SequenceType type = GetSequenceType();
             if (type == SequenceType.MultiSegment && startObject != endObject)
             {
-                Debug.Assert(startObject != null);
-                Debug.Assert(startObject is ReadOnlySequenceSegment<T>);
-                Debug.Assert(endObject != null);
-                Debug.Assert(endObject is ReadOnlySequenceSegment<T>);
-
-                var startSegment = Unsafe.As<ReadOnlySequenceSegment<T>>(startObject);
-                var endSegment = Unsafe.As<ReadOnlySequenceSegment<T>>(endObject);
+                var startSegment = (ReadOnlySequenceSegment<T>)startObject;
+                var endSegment = (ReadOnlySequenceSegment<T>)endObject;
 
-                // start.RunningIndex + startIndex <= end.RunningIndex + endIndex
```

Comment:
Why remove this comment along with `// Rearranged to avoid overflow`?
What does `// Length >= 0` mean?

Reply:
`// Length >= 0` means that bounds are correct. I can extend message in comment.